test/net_http_test.rb

```diff
@@ -42,6 +42,40 @@ def test_semian_identifier
     end
   end
 
+  def test_changes_timeout_when_half_open_and_configured
```

Comment:
I'm not sure if I'm doing this right, is there a better way of half opening the circuit?

Reply:
I think I'll need help with this test, I can't seem to be able to trigger the right thing. I do hit the `with_resource_timeout` path but `ReadTimeout` does not raise.